### PR TITLE
Add Cloud Platform scope to NodePool Creation by default on Pathways + CPU

### PR DESCRIPTION
**Pull request by SujeethJinesh**
## Fixes / Features
- This PR adds the Cloud Platform access scope to keep consistency between pathways and McJAX NodePool creation. This will get rid of the warning below and make it easy for us to get goodput values by default. 

Screenshot 2024-11-07 at 2 47 22 PM

## Testing / Documentation
N/A

Testing details.

- [ y ] Tests pass
- [ y ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

**Review by dipannita08 — APPROVED** [review]
Thanks!

**Comment by pawloch00** [comment]
Please use cloud-platform instead of url

File: src/xpk/core/core.py [hunk 1]
```diff
@@ -1500,7 +1500,7 @@ def run_gke_node_pool_create_command(
           f' --region={zone_to_region(args.zone)}'
           ' --num-nodes=1'
           f' --machine-type={args.pathways_gce_machine_type}'
-          ' --scopes=storage-full,gke-default'
+          ' --scopes=storage-full,gke-default,"https://www.googleapis.com/auth/cloud-platform"'
```
Comment (pawloch00) [inline src/xpk/core/core.py:1503]: Please use cloud-platform instead of url

Comment (SujeethJinesh) [reply src/xpk/core/core.py:1503]: We currently use the full cloud-platform link in other parts of the same file. Should we keep it consistent or replace it (need to verify `cloud-platform` on its own also works).

E.g. 
https://github.com/AI-Hypercomputer/xpk/blob/main/src/xpk/core/core.py#L1449

and

https://github.com/AI-Hypercomputer/xpk/blob/main/src/xpk/core/core.py#L1460

**Review by shauryagup — APPROVED** [review]

File: src/xpk/core/core.py (hunk 1 above)
Comment (SujeethJinesh) [reply src/xpk/core/core.py]: Done. Replaced with constant URL instead. Thanks!

**Review by pawloch00 — APPROVED** [review]

**Comment by pawloch00** [comment]
Does warning you pasted in Readme occurs to every user using Pathways, or just in your case? (Sorry, I am not good with pathways yet)

**Comment by SujeethJinesh** [comment]
> Does warning you pasted in Readme occurs to every user using Pathways, or just in your case? (Sorry, I am not good with pathways yet)

No worries! This issue would appear for anytime goodput is enabled and the cloud-platform access scope is not present, which is a great reminder and thanks @dipannita08 for pointing it out as well, that we should make this scope default for CPU accelerator types as well (added now). Thanks! I'll need another LGTM.

**Review by dipannita08 — APPROVED** [review]

**Review by pawloch00 — APPROVED** [review]

**Review by dipannita08 — APPROVED** [review]

**Pull request merged**